src/electionguard/random.hpp

```diff
@@ -17,24 +17,60 @@ using namespace std;
 
 namespace electionguard
 {
-    enum ByteSize { SHA256 = 256, SHA384 = 384, SHA512 = 512 };
+    enum ByteSize {
+        CHAR = 8,
+        SHORT = 16,
+        INT = 32,
+        LONG = 64,
+        SHA256 = 256,
+        SHA384 = 384,
+        SHA512 = 512
+    };
+
+    /// <summary>
+    /// A convenience wrapper around the HACL* HMAC Deterministic Random Bit Generator
+    /// that supports retrieving an arbitrary number of pseudo-random bytes as specified.
+    ///
+    /// This implementation currently does not support reseeding the same instantiation
+    /// and expects the caller to initialize a new instance before the entropy pool is exhausted
+    ///
+    /// Please refer to the NIST publication for more information:
+    /// https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf
+    /// </summary>
     class Random
     {
       public:
-        /// https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf
+        /// <summary>
+        /// Get pseudo-random bytes for the specified ByteSize.
+        ///
+        /// Note:
+        /// This implmenetation expects the OS random source to have at least `size * 2`
```

Comment:
i didnt want to get too specific with the comment, but this should not be an issue on any of our target hardware.  If, for instance someone tried to run this on an ultra low powered microcontroller its possible the entropy source may be exhausted depending on what other tasks the operating system is executing.  there's no good way for us to control for that, so consumers just need to know that dev/urandom guarantees a quick response, but not high quality bits.